doc/format.rst

```diff
@@ -514,3 +498,46 @@ The ``neural_networks`` section is required and must define the following:
 -  ``static`` [BOOL, REQUIRED]: The hybridization type
    (see :ref:`hybridization types <hybrid_types>`). ``true`` indicates
    static hybridization; ``false`` indicates dynamic hybridization.
+
+Notes for developers
+--------------------
+
+This section outlines recommendations and tips for developers interested in adding PEtab
+SciML support to their packages.
+
+Alternative model and neural-network formats
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Both the ODE model and NN formats are flexible. Still, the most widely supported model
+format is `SBML <https://sbml.org/>`_, the de facto standard for dynamical models in
+computational biology (field of standard developers). We recommend supporting SBML whenever
+possible to promote model exchange. Likewise, we recommend supporting the PEtab SciML
+NN :ref:`YAML format <NN_YAML>`.
+
+That said, alternative model formats (e.g., `BioNetGen <https://bionetgen.org/>`_) or
+language-specific formulations, and alternative NN formats (e.g., architectures not yet
+covered by the YAML format), may suit some tools, especially outside biology. The PEtab
+SciML standard remains useful across formats by providing a high-level abstraction that
+connects the dynamical model and NN components regardless of representation. For example,
+leveraging this abstraction, PEtab.jl provides a Julia interface to create the PEtab tables
+and can accept a `DifferentialEquations.jl <https://diffeq.sciml.ai/>`_ ``ODEProblem`` as
+the model together with NNs defined in `Lux.jl <https://lux.csail.mit.edu/>`_. If
+adding support for other formats, to **thoroughly test correctness**, the PEtab SciML
+`test suite <https://github.com/sebapersson/petab_sciml_testsuite>`_ can be adapted by
+replacing the NN and/or model files to match the formats any importer targets.
+
+Dealing with arrays
+~~~~~~~~~~~~~~~~~~~
+
+For array handling, it is recommended to:
+
+- **Respect memory/axis order.** For computational efficiency, reorder input data and any
+   layer-parameter arrays to the target language’s native indexing and memory layout.
+   For example, PEtab.jl permutes image inputs to Julia’s ``(H, W, C)`` convention.
+
+- **Allow export of parameters in PEtab SciML format**. If the NN model is not provided in
+   the YAML format, exchange of NN parameters between software is not possible as then
+   the parameter should following the indexing and memory layout of the respective
+   framework. To facilitate exchange, it is recommended that tools supporting PEtab SciML
+   implement a function  capable of exporting to the PEtab SciML array data format with
+   PyTorch indexing.
```

Comment:
I don't understand this point -- is it that, if the NN model is not provided in YAML format, then the array data files cannot be interpreted? I thought that array data files could be interpreted regardless (up to tools to implement them for non-YAML NN models).